apps/client/src/features/rundown/RundownEntry.tsx

```diff
@@ -31,27 +41,36 @@ interface RundownEntryProps {
 }
 
 export default function RundownEntry(props: RundownEntryProps) {
-  const { isPast, data, selected, hasCursor, next, previousEnd, previousEventId, playback, isRolling, disableEdit } =
-    props;
+  const {
+    isPast,
+    data,
+    selected,
+    hasCursor,
+    next,
+    previousEnd,
+    previousEventId,
+    playback,
+    isRolling,
+    disableEdit,
+    eventIndex,
+  } = props;
   const { emitError } = useEmitLog();
-  const { addEvent, updateEvent, deleteEvent, swapEvents } = useEventAction();
-
-  const cursor = useAppMode((state) => state.cursor);
-  const setCursor = useAppMode((state) => state.setCursor);
-  const openId = useAppMode((state) => state.editId);
-  const setEditId = useAppMode((state) => state.setEditId);
+  const { addEvent, updateEvent, batchUpdateEvents, deleteEvent, swapEvents } = useEventAction();
+  const { data: rundown = [] } = useRundown();
+  const { cursor } = useAppMode();
+  const { eventsToEdit, clearEventsToEdit } = useEventSelection();
 
   const removeOpenEvent = useCallback(() => {
-    if (openId === data.id) {
-      setEditId(null);
+    if (eventsToEdit.has(data.id)) {
+      clearEventsToEdit();
     }
 
     if (cursor === data.id) {
-      setCursor(null);
+      // setCursor(null);
     }
-  }, [cursor, data.id, openId, setCursor, setEditId]);
+  }, [cursor, data.id, eventsToEdit, clearEventsToEdit]);
 
-  const eventSettings = useEditorSettings((state) => state.eventSettings);
+  const { eventSettings } = useEditorSettings();
```

Comment:
same as before: I believe it would be best to use the memoised selector syntax

Reply:
Resolved with https://github.com/cpvalente/ontime/pull/497/commits/da1b32811475028b5d8d42323e9ad678255913a1